types/mithril/index.d.ts

```diff
@@ -171,7 +171,7 @@ declare namespace Mithril {
 	type ChildArrayOrPrimitive = ChildArray | string | number | boolean;
 
 	/** Virtual DOM nodes, or vnodes, are Javascript objects that represent an element (or parts of the DOM). */
-	interface Vnode<Attrs, State extends Lifecycle<Attrs, State>> {
```

Comment:
This is a required constraint to permit certain optional methods to be implemented (hooks like `onremove` for when the underlying DOM node is removed), so if there's another way of fixing this, please do.